dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -551,7 +619,79 @@ public void test_MultilangContent_IndexingAllLanguages()
             SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catalina", 0, 10);
             Assert.assertTrue(searchResults.getTotalResults() >= 1);
 
-            searchResults = siteSearchAPI.search(newIndexName, "catherine", 0, 10);
+        } finally {
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE",
+                    defaultContentToDefaultLangOriginalValue);
+        }
+    }
+
+    /**
+     * Given sceneario: Content in default language only referenced from a page in default language, and having DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true.
+     * Create a Site-Search run including the second language only.
+     * Expected result: the page should be in the results, because DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true and DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true .
+     */
+
+    @Test
+    public void test_DefaultLangContent_DefaultLangPage_IndexingSecondLang_ShouldFindResults()
+            throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
+
+        boolean defaultContentToDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+
+        try {
+
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", true);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
+            folder = new FolderDataGen().site(site).nextPersisted();
+
+            Contentlet contentDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
+
+            // create container, template and htmlpage to reference previously created content
+
+            HTMLPageAsset page = createHtmlPageAsset(defaultLang, contentDefaultLang);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'page' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)